hkdf/tests/tests.rs

```diff
@@ -91,17 +92,29 @@ fn test_rfc5869_sha256() {
         } else {
             Some(&salt[..])
         };
+
         let (prk2, hkdf) = Hkdf::<Sha256>::extract(salt, ikm);
         let mut okm2 = vec![0u8; okm.len()];
         assert!(hkdf.expand(&info[..], &mut okm2).is_ok());
-
         assert_eq!(prk2[..], prk[..]);
         assert_eq!(okm2[..], okm[..]);
 
-        okm2.iter_mut().for_each(|b| *b = 0);
+        okm2.iter_mut().zeroize();
```

Comment:
My suspicion is that `okm2.zeroize()` will work directly---i.e. you may not need the `iter_mut()`. If so this will apply to all instances

Reply:
I had that initially, but it turned out to clear the `Vec`, i.e. 0 elements.

Reply:
Specifically, it calls `iter_mut.zeroize()` then `clear()`: https://github.com/RustCrypto/utils/blob/73ad6a6fb26e30483580f71c98b901d27803565e/zeroize/src/lib.rs#L469

Reply:
i think for slices it does zeroize -- i didn't know it clears Vec though